aya/src/maps/perf/async_perf_event_array.rs

```diff
@@ -98,16 +98,17 @@ impl<T: BorrowMut<MapData> + Borrow<MapData>> AsyncPerfEventArray<T> {
         index: u32,
         page_count: Option<usize>,
     ) -> Result<AsyncPerfEventArrayBuffer<T>, PerfBufferError> {
-        let buf = self.perf_map.open(index, page_count)?;
+        let Self { perf_map } = self;
+        let buf = perf_map.open(index, page_count)?;
         let fd = buf.as_raw_fd();
         Ok(AsyncPerfEventArrayBuffer {
             buf,
 
             #[cfg(feature = "async_tokio")]
-            async_fd: AsyncFd::new(fd)?,
+            async_tokio_fd: AsyncFd::new(fd)?,
```

Comment:
I'm not sure I understand - is the purpose of this change to make cargo hack
happy? This code (pre PR) is messy and I'm glad it's getting improved, but now
it'll let you compile with both features on but won't do the right thing at
runtime. I think we should compile_error or something if both features are on?

Also I'm not sure that merging the `read_events` implementations is an
improvement. The code seems harder to read and understand to me. Not having to
write docs twice is an improvement, but that can probably be achieved in some
other way.

Reply:
We can compile error, but we're in somewhat unsupported territory because mutually exclusive features aren't really a thing in cargo.

We can also do something approximating the old code: pick tokio If both are enabled.

A third option is to crash at runtime. Not clear to me what's best here.

Reply:
I think we should keep the current behaviour and use tokio when both are enabled https://doc.rust-lang.org/cargo/reference/features.html#mutually-exclusive-features

Reply:
Done.